axiom/nr_errors.h

```diff
@@ -38,6 +38,38 @@ extern nr_error_t* nr_error_create(int priority,
                                    const char* span_id,
                                    nrtime_t when);
 
+/*
+ * Purpose : Create a new error for the use case where additional parameters are
+ * passed in.
+ *
+ * Params  : 1. The importance of the error.  Higher number is more important.
+ *           2. Message string.
+ *           3. Class string.  This string is used to aggregate errors in RPM.
+ *              RPM does not expect strings from a specific namespace.
+ *              The PHP agent uses PHP's predefined error constant names.
+ *              Examples are "E_ERROR" and "E_WARNING".
+ *           4. Error file provided by user.
+ *           5. Error line provided by user.
+ *           6. Error context provided by user.
+ *           7. Error number provided by user.
+ *           8. String containing stack trace in JSON format.
+ *           9. Span ID
+ *           10. When the error occurred.
+ *
```

Comment:
Would be good to explain if all are required or if there are certain combinations allowed.

Reply:
Added: [9fad7fa](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/9fad7fae54e3fa1a4a2d2dfecf7493e61ee1f2a3)